modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.downloadsButtonPlugin/mh_downloads.js

```diff
@@ -0,0 +1,272 @@
+        self.loadTracksFromEpisode();
+      }
+      self.renderTracks(domElement);
+    }
+
+    loadTracksFromEpisode() {
+      var self = this;
+      var tracks = self._episode.mediapackage.media.track;
+      if (!(tracks instanceof Array)) { tracks = [tracks]; }
+
+      var parsedTracks = [];
+      for (var tracksIndex = 0; tracksIndex < tracks.length; tracksIndex++) {
+        var track = tracks[tracksIndex];
+        // only show static file content over http
+        // skip streaming tracks
+        if (track.url.startsWith('http')
+            && (track.mimetype.indexOf('video') >= 0) || track.mimetype.indexOf('audio') >= 0) {
+
+          // skip live events
+          if (track.live != undefined) {
+            if (track.live) { continue; }
+          }
+
+          // skip and tags
+          if (self._tags.length > 0 && (track.tags == undefined || track.tags.tag == undefined
+                || track.tags.tag.length < 1)) {
+            continue;
+          }
+
+          var match_tags = false;
+          for (var tagsIndex = 0; tagsIndex < self._tags.length; tagsIndex++) {
+            var tag = self._tags[tagsIndex];
+            if (track.tags.tag.indexOf(tag) >= 0) {
+              match_tags = true;
+              break;
+            }
+          }
+          // skip by flavor
+          var match_flavors = false;
+          for (var flavorsIndex = 0; flavorsIndex < self._flavors.length; flavorsIndex++) {
+            var flavor = self._flavors[flavorsIndex];
+            if (track.type === flavor) {
+              match_flavors = true;
+              break;
+            }
+          }
+
+          if (self._tags.length > 0 && self._flavors.length > 0) {
+            if (self._tags_and_flavors) {
+              if (!(match_tags && match_flavors)) { continue; }
+            } else {
+              if (!(match_tags || match_flavors)) { continue; }
+            }
+          } else {
+            if (self._tags.length > 0 && !match_tags) { continue; }
+            if (self._flavors.length > 0 && !match_flavors) { continue; }
+          }
+
+          var parsedTrack = new DownloadTrack(track);
+          parsedTracks.push(parsedTrack);
+        }
+      }
+      self._tracks = parsedTracks;
+    }
+
+    renderTracks(domElement) {
+      var self = this;
+      let container = document.createElement('div');
+      container.className = 'downloadsButtonContainer';
+
+      if (self._episode === undefined || self._tracks.length < 1) {
+        container.innerHTML = '<div class="downloadsItemContainer"><p class="noTracks">'
+                                + paella.dictionary.translate('No downloads available') + '</p></div';
+      } else {
+        let trackTypes = ['video', 'audio'];
+        let flavors = self._tracks.reduce(function(result, track) {
+          if (result.indexOf(track.flavor) < 0) {
+            result.push(track.flavor);
+          }
+          return result;
+        }, []);
```

Comment:
It might be simpler to use sets instead:

```suggestion
        let flavors = new Set(self._tracks.map(track => track.flavor)); 
```

But that also means that you would need to use Set functions below and iterate using for-of of forEach